src/group/managed_group/resumption.rs

```diff
@@ -0,0 +1,45 @@
+use super::*;
+
+/// Resumption secrets store. This is where the resumption secrets are kept in a rollover list.
+#[derive(Debug, Serialize, Deserialize)]
+#[cfg_attr(test, derive(PartialEq))]
+pub(crate) struct ResumptionSecretStore {
+    number_of_secrets: usize,
+    resumption_secrets: Vec<(GroupEpoch, ResumptionSecret)>,
+    cursor: usize,
+}
+
+impl ResumptionSecretStore {
+    /// Create e new store with a given maximum size of `number_of_secrets`.
+    pub(crate) fn new(number_of_secrets: usize) -> Self {
+        Self {
+            number_of_secrets,
+            resumption_secrets: vec![],
+            cursor: 0,
+        }
+    }
+
+    /// Add a new entry to the store.
+    pub(crate) fn add(&mut self, epoch: GroupEpoch, resumption_secret: ResumptionSecret) {
```

Comment:
I guess that could work as well, although it feels more brittle. I thought the linear effort for searching in the resumption keys was tolerable, since this is not called very often and we don't expect a huge number of keys.
If you think we need this performance at the expense of robustness (and possibly hidden cost to initialise the `HashMap` whenever we load the state, we can discuss.